libs/base58_nostd/src/lib.rs

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) 2021 RBB S.r.l
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+// Based on https://github.com/trezor/trezor-crypto/blob/master/base58.c
+// commit hash: c6e7d37
+// license: MIT
+// works only up to 128 bytes
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::sp_io::hashing::sha2_256;
+use sp_std::vec;
+use sp_std::vec::Vec;
+
+pub const TOKEN_ID_PREFIX: &'static str = "MLS";
```

Comment:
1 char for prefix is good enough. We can engineer a better prefix later with bruteforce.

Reply:
To clarify: Obviously, MLS will not show us in your base58check result. This is because MLS is taken as binary data.

Reply:
I fixed it. Now, we use just `M`